### ISSUE TITLE
Bug: no-unnecessary-condition: false positive on unnecessary optional chain

### ISSUE THREAD

**Issue by smaye81** [typescript-eslint/typescript-eslint#8022]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

I expected that the code should pass lint as it is clear that `Options` could be undefined, therefore the optional chain is necessary. However, it fails saying 'unnecessary condition' and even more curious, fails on the _second_ instance of the optional chain, ignoring the first.

### Reproduction Repository Link

https://github.com/smaye81/eslint-repro

### Repro Steps

1) clone the repo
2) `npm install`
3) `npm run lint`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.13.1` |
| `@typescript-eslint/parser`            | `6.13.1` |
| `TypeScript`                           | `5.3.2` |
| `ESLint`                               | `8.55.0` |

**Comment by Josh-Cena** [comment]
You can simply use [the playground](https://typescript-eslint.io/play#ts=5.3.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgPIAczAPYgM7IDeAUMsjMFHmAPwBcyARttgDYRwgDcpyeECXABN6TFu048AvsWIQAHumxQw5AK4gEWXMkjUAFNkw58DDNvzIAPsg1CIFEBCEBKIr2Axkh47jw0AOgoqVQBCAF5w8jhWfjcSMjJBfDYIANZsAHN9ACJg6hyXHjIZMgB6MuQAGVBVUEEoKAEwVgBPZEEAW3RWOFACMAALOFUh4AJx5E5bECckPDw4KHbkoWALXgqmNVV%2BCE6B7CYURxQAd3XB5FZQFABOAI8vHwt-AP5V5Aio%2BFiIeN4ST8qXSWVyH2EhWKyBkMmIQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false) instead of a personal repo, by the way. Both easier for you to set up and easier for us to look at.

This is a TypeScript bug because TypeScript thinks `options` is non-null there. If you remove the optional chain, you see that TypeScript doesn't complain either about "'options' is possibly 'undefined'".

**Comment by Josh-Cena** [comment]
Fixed in https://github.com/microsoft/TypeScript/issues/56482

**Comment by smaye81** [comment]
I tried the playground actually, but was unable to change the ESLint version. Not sure if I'm missing something.

Also re: TypeScript, if I remove that optional chain, then `tsc` fails.

**Comment by smaye81** [comment]
Ah great, thanks @Josh-Cena

**Comment by smaye81** [comment]
Got it thanks.

Also definitely running the 5.3 version:

```bash
$ npx tsc --version                                                                                                                                                                
Version 5.3.2

$ npx tsc --noEmit                                                                                                                                                                                                                                                                                                                                                                
index.ts:9:7 - error TS18048: 'options' is possibly 'undefined'.

9   if (options.first !== false) {
        ~~~~~~~

Found 1 error in index.ts:9
```

FWIW, I tried this on the latest 5.4.0 TypeScript nightly and it also fails there. Maybe the 5.3.3 fix just hasn't been merged into that?

**Comment by Josh-Cena** [comment]
Ah, you need to keep the first optional chain but remove the second ;)

```ts
interface Options {
  first?: boolean;
  second?: boolean;
}

export function test(options: Options | undefined) {
  if (options?.first !== false) {
    console.log("first");
  }
  if (options.second !== false) {
    console.log("second");
  }
}
```

**Comment by smaye81** [comment]
Even that seems wrong, though, right? In my code example, the `options.second` line will always run and `options` could be `undefined` there?

**Comment by Josh-Cena** [comment]
Right, hence it's a TS bug!

**Comment by bradzacher** [comment]
Also duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/7977
